### PR TITLE
Add export output names specification for individual nodes

### PR DESCRIPTION
**Pull request by sokovninn**
This pull request includes changes to deprecate the `output_names` configuration in favor of `export_output_names` for more granular control. It updates the documentation, configuration files, and relevant code to reflect this change.

### Deprecation of `output_names`:

* [`configs/README.md`](diffhunk://#diff-26db36ffc817266dc58ecfee77c2d48a7b42090938e644ab8bf35b4696f3e3e3L391-R391): Marked `output_names` as deprecated in the documentation.
* [`luxonis_train/models/luxonis_lightning.py`](diffhunk://#diff-81464460706d481358131ac236b9102acacfcc172680c2b8f976dc70ba37e7e1R523-R554): Added a warning message about the deprecation of `exporter.output_names` and implemented logic to use `node.export_output_names` if provided. [[1]](diffhunk://#diff-81464460706d481358131ac236b9102acacfcc172680c2b8f976dc70ba37e7e1R523-R554) [[2]](diffhunk://#diff-81464460706d481358131ac236b9102acacfcc172680c2b8f976dc70ba37e7e1R575-R581)

### Configuration updates:

* [`configs/example_export.yaml`](diffhunk://#diff-6fe37fe278ac632dc5b25f81091c2789216142578dd5a89049cf3129f11577ccR10-R11): Removed `output_names` and added `export_output_names` under `head_params`. [[1]](diffhunk://#diff-6fe37fe278ac632dc5b25f81091c2789216142578dd5a89049cf3129f11577ccR10-R11) [[2]](diffhunk://#diff-6fe37fe278ac632dc5b25f81091c2789216142578dd5a89049cf3129f11577ccL45)
* [`tests/configs/archive_config.yaml`](diffhunk://#diff-f8bf6e01fff53ac1f5ce162552749eeb5cd7ed58fb15c3d5dcfc8074d4896e29R10-R35): Updated the configuration to use `export_output_names` for various nodes.

### Code changes:

* [`luxonis_train/nodes/base_node.py`](diffhunk://#diff-d2c6ee7bd11bfb00ba122300c93a3127e45450548f2022228c482465e35f358eR122): Added `export_output_names` as an attribute and updated the constructor and property methods accordingly. [[1]](diffhunk://#diff-d2c6ee7bd11bfb00ba122300c93a3127e45450548f2022228c482465e35f358eR122) [[2]](diffhunk://#diff-d2c6ee7bd11bfb00ba122300c93a3127e45450548f2022228c482465e35f358eR143-R147) [[3]](diffhunk://#diff-d2c6ee7bd11bfb00ba122300c93a3127e45450548f2022228c482465e35f358eR195) [[4]](diffhunk://#diff-d2c6ee7bd11bfb00ba122300c93a3127e45450548f2022228c482465e35f358eR523-R527)
* [`luxonis_train/nodes/heads/efficient_bbox_head.py`](diffhunk://#diff-976671347651a5d336e4f712bcf9afc07ae0f24be5907a71dac72d4ead40dcf7R77-R92): Added logic to handle `export_output_names` in the constructor.

### CODE REVIEW

File: luxonis_train/models/luxonis_lightning.py [hunk 1]
```diff
@@ -520,7 +520,38 @@ def export_onnx(self, save_path: str, **kwargs) -> list[str]:
...
 
+        output_counts = defaultdict(int)
+        for node_name, outs in outputs.items():
+            output_counts[node_name] = sum(len(out) for out in outs.values())
+
         if self.cfg.exporter.output_names is not None:
+            logger.warning(
+                "The use of 'exporter.output_names' is deprecated and will be removed in a future version. "
+                "If 'node.export_output_names' are provided, they will take precedence and overwrite 'exporter.output_names'. "
+                "Please update your code to use 'node.export_output_names' directly."
```
Comment (klemen1999) [inline luxonis_train/models/luxonis_lightning.py:531]:
```suggestion
                "Please update your config to use 'node.export_output_names' directly."
```

**Review by klemen1999 — APPROVED** [review]
LGTM. I guess the code will be much nicer after the old way will be deprecated

**Comment by sokovninn** [comment]
@klemen1999 @kozlov721 archive generation for models with multiple heads should work well now. Please take a look.

**Review by klemen1999 — APPROVED** [review]
LGTM

**Review by kozlov721 — APPROVED** [review]
LGTM

**Pull request merged**